examples/app.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-License-Identifier: MIT
+# Copyright (C) 2024 Avnet
+# Authors: Nikola Markovic <nikola.markovic@avnet.com> et al.
+# This is a self-updating app with OTA support that can update itself with a new OTA package.
+
+import random
+import sys
+import time
+from dataclasses import dataclass
+import subprocess
+import os
+import urllib.request
+
+
+from avnet.iotconnect.sdk.lite import Client, DeviceConfig, C2dCommand, Callbacks, DeviceConfigError
+from avnet.iotconnect.sdk.lite import __version__ as SDK_VERSION
+from avnet.iotconnect.sdk.sdklib.mqtt import C2dAck, C2dOta
+
+"""
+See minimal.py example for a way to configure the device without the JSON file
+You can download the iotcDeviceConfig.json by clicking the cog icon in the upper right of your device's info panel
+NOTE: If you do not pass the server certificate, we will use the system's trusted certificate store, if available.
+For example, the trusted Root CA certificates from the in /etc/ssl/certs will be used on Linux.
+However, it is more secure to pass the actual server CA Root certificate in order to avoid potential MITM attacks.
+On Linux, you can use server_ca_cert_path="/etc/ssl/certs/DigiCert_Global_Root_CA.pem" for Azure
+or server_ca_cert_path="/etc/ssl/certs/Amazon_Root_CA_1.pem" for AWS
+"""
+
+@dataclass
+class ExampleAccelerometerData:
+    x: float
+    y: float
+    z: float
+
+
+@dataclass
+class ExampleSensorData:
+    temperature: float
+    humidity: float
+    accel: ExampleAccelerometerData
+
+
+def on_command(msg: C2dCommand):
+    print("Received command", msg.command_name, msg.command_args, msg.ack_id)
+    if msg.command_name == "set-user-led":
+        if len(msg.command_args) == 3:
+            status_message = "Setting User LED to R:%d G:%d B:%d" % (int(msg.command_args[0]), int(msg.command_args[1]), int(msg.command_args[2]))
+            c.send_command_ack(msg, C2dAck.CMD_SUCCESS_WITH_ACK, status_message)
+            print(status_message)
+        else:
+            c.send_command_ack(msg, C2dAck.CMD_FAILED, "Expected 3 arguments")
+            print("Expected three command arguments, but got", len(msg.command_args))
+    else:
+        print("Command %s not implemented!" % msg.command_name)
+        if msg.ack_id is not None: # it could be a command without "Acknowledgement Required" flag in the device template
+            c.send_command_ack(msg, C2dAck.CMD_FAILED, "Not Implemented")
+
+
+def on_ota(msg: C2dOta):
+    print("Starting OTA downloads for version %s" % msg.version)
+    error_msg = None
+    c.send_ota_ack(msg, C2dAck.OTA_DOWNLOADING)
```

Comment:
Incorporate ota-download.py example into the logic. Support different kinds of files. Ota-handling also has some threading incorporated for a reason. Investigate the reason and see if it applies to this as well.